account_invoice_view_payment/models/account_move.py

```diff
@@ -0,0 +1,94 @@
+        self.ensure_one()
+        reconciled_vals = []
+        for move in self:
+            if move.state == "posted" and move.is_invoice(include_receipts=True):
+                reconciled_partials = move._get_all_reconciled_invoice_partials()
+                for reconciled_partial in reconciled_partials:
+                    counterpart_line = reconciled_partial["aml"]
+                    if counterpart_line.move_id.ref:
+                        reconciliation_ref = "%s (%s)" % (
+                            counterpart_line.move_id.name,
+                            counterpart_line.move_id.ref,
+                        )
+                    else:
+                        reconciliation_ref = counterpart_line.move_id.name
+                    if (
+                        counterpart_line.amount_currency
+                        and counterpart_line.currency_id
+                        != counterpart_line.company_id.currency_id
+                    ):
+                        foreign_currency = counterpart_line.currency_id
+                    else:
+                        foreign_currency = False
+
+                    reconciled_vals.append(
+                        {
+                            "name": counterpart_line.name,
+                            "journal_name": counterpart_line.journal_id.name,
+                            "amount": reconciled_partial["amount"],
+                            "currency_id": move.company_id.currency_id.id
+                            if reconciled_partial["is_exchange"]
+                            else reconciled_partial["currency"].id,
+                            "date": counterpart_line.date,
+                            "partial_id": reconciled_partial["partial_id"],
+                            "account_payment_id": counterpart_line.payment_id.id,
+                            "payment_method_name": counterpart_line.payment_id.payment_method_line_id.name,  # noqa: B950
+                            "move_id": counterpart_line.move_id.id,
+                            "ref": reconciliation_ref,
+                            # these are necessary for the views to change
+                            # depending on the values
+                            "is_exchange": reconciled_partial["is_exchange"],
+                            "amount_company_currency": formatLang(
+                                self.env,
+                                abs(counterpart_line.balance),
+                                currency_obj=counterpart_line.company_id.currency_id,
+                            ),
+                            "amount_foreign_currency": foreign_currency
+                            and formatLang(
+                                self.env,
+                                abs(counterpart_line.amount_currency),
+                                currency_obj=foreign_currency,
+                            ),
+                        }
+                    )
+        return reconciled_vals
```

Comment:
**issue**: I don't think we should reintroduce this function. It changed name and structure slightly, but it's `_compute_payments_widget_reconciled_info` in v16 and I think you should be able to use it in the same way.

Reply:
Introduced new function as reconciled_vals is not obtainable from existing _compute_payments_widget_reconciled_info() or any other available methods in v16.

However, account_invoice_view_payment only uses payment related information hence using get_reconciled_payments method to achieve the same in order to view payments from invoices.

Reply:
ah, nice catch!